util_function_groups/column_operations.R

```diff
@@ -3,6 +3,219 @@
 source("~/Sites/gymnast/util.R", chdir = FALSE)
 source("~/Sites/gymnast/util_qualtrics_cleaning.R", chdir = FALSE)
 
+
+############################### rename_columns function  #################################################
+
+rename_columns <- function(data, mapping) {
+  # This function takes a data set and a mapping (which can be a list or data frame)
+  # with columns "old_name" and "new_name". It uses util.recode to turn the old names
+  # into the new names, making sure not to break any R name rules.
+  
+  # Patterns for forbidden names
+  whitespace_pattern <- "\\s"
+  starting_w_digit_pattern <- "^\\.?[0-9]"
+  naughty_words <- c("if", "else", "repeat", "while", "function", "for",
+                     "in", "next", "break", "TRUE", "FALSE", "NULL", "Inf",
+                     "NaN", "NA", "NA_integer_", "NA_real_", "NA_complex_", "NA_character_", NA)
+  
+  # if the new names in the mapping match any evil patterns, then complain
+  if( any(mapping$new_name %in% naughty_words) |
+      any(grepl(whitespace_pattern, mapping$new_name)) |
+      any(grepl(starting_w_digit_pattern, mapping$new_name))) {
+    stop("Invalid new names are present among: ",paste(mapping$new_name, collapse =" "))
```

Comment:
@dkgreene messy whitespace here and elsewhere, please go over the file to check.

For example, this line should be:

`   stop("Invalid new names are present among: ", paste(mapping$new_name, collapse = " "))`